transport_generic.go

```diff
@@ -26,10 +42,11 @@ func (t genericTransport) ReadMessage() (*Message, error) {
 }
 
 func (t genericTransport) SendMessage(msg *Message) error {
+	nativeEndian = NativeEndian()
```

Comment:
It would be better to do this in the 'init' function for the package so that 'nativeEndian' will be populated when the package is initialized. Another way is to create transport_arch.go files that define the byte order for the different architectures so it doesn't need to be detected, but that seems brittle.

Something like:
```
var nativeEndian binary.ByteOrder
func detectEndianness() binary.ByteOrder {
	if nativeEndian == nil {
		var x uint32 = 0x01020304
		if *(*byte)(unsafe.Pointer(&x)) == 0x01 {
			nativeEndian = binary.BigEndian
		} else {
			nativeEndian = binary.LittleEndian
		}
	}
	return nativeEndian
}
func init() {
        nativeEndian = detectEndianness()
}
```
then you can just use the 'nativeEndian' variable where you need it.